File: problems/custom/matlab_problems/custom1.m
function problem = custom1()
    % This is a toy example to show how to construct a .m file that returns
    % a instance of Problem class representing an unconstrained problem.

    p_struct.name = 'custom1';
    p_struct.x0 = [1; 1];
    p_struct.fun = @fun;

    problem = Problem(p_struct);
end

function f = fun(x)
    f = sum(x.^2);
end

File: problems/custom/python_problems/custom1.py
def custom1():
    """
    This is a toy example to show how to construct a .py file that returns a
    dictionary that describes an unconstrained optimization problem.
    """
    p_dict = {}
    p_dict['name'] = 'custom1'
    p_dict['x0'] = [1.0, 1.0]
    p_dict['fun'] = fun

    return p_dict

def fun(x):
    return sum(xi**2 for xi in x)